### PR TITLE
Integrate Intercom and handle incoming messages

### PR DESCRIPTION
**Pull request by biafra23**
Handles #1656 

Changes proposed in this pull request:
- Integrated Intercom SDK
- Chat can be entered via `App Settings` -> `Chat with us`
- Unread replies from support are highlighted on the Settings Icon in the tab bar and the App Settings item `Chat with us`

image image

The badge in the tab bar is too big. But we could not find an easy way to make it smaller. Any ideas?

Paired with @sche

### CODE REVIEW

File: Multisig/UI/App/MainTabBarViewController.swift [hunk 1]
```diff
@@ -154,16 +161,37 @@ class MainTabBarViewController: UITabBarController {
...
     }
 
+    @objc func showBadge() {
+        let count = Intercom.unreadConversationCount()
+        LogService.shared.debug("showBadge() called: count: \(count), settingsTabItem:   \(settingsTabItem)")
+        if count > 0 {
+            //TODO: Find out what value just shows a badge without a number
+            settingsTabItem?.badgeValue = "\(count)"
+            //TODO: How to set to apps primary color?
+            settingsTabItem?.badgeColor = UIColor.green
```
Comment (biafra23) [inline Multisig/UI/App/MainTabBarViewController.swift]: How should I reference the Safes primary color?

File: Multisig/UI/App/MainTabBarViewController.swift [hunk 2]
```diff
@@ -154,16 +161,37 @@ class MainTabBarViewController: UITabBarController {
...
-    private func tabViewController(root: UIViewController, title: String, image: UIImage, tag: Int) -> UIViewController {
+    //TODO: Find a better way to update tab than storing a reference here
+    private var settingsTabItem : UITabBarItem? = nil
+
+    private func tabViewController(root: UIViewController, title: String, image: UIImage, tag: Int, isSettingsTab: Bool = false) -> UIViewController {
         let nav = UINavigationController(rootViewController: root)
         let tabItem = UITabBarItem(title: title, image: image, tag: tag)
+
+        //TODO: Can we not use this haclk to have a reference to the UITabBarItem?
+        if isSettingsTab {
```
Comment (biafra23) [inline Multisig/UI/App/MainTabBarViewController.swift]: When we receive an NSNotification about an unread conversation we want to update the badge of the settings tab bar item. What is the best way to get a reference to it?  To be used in `showBadge()` later.

File: Multisig/UI/Settings/AppSettingsViewController/AppSettingsViewController.swift [hunk 3]
```diff
@@ -100,7 +106,7 @@ class AppSettingsViewController: UITableViewController {
                 Section.App.experimental("Experimental")
             ]),
             (section: .support("Support & Feedback"), items: [
-                Section.Support.chatWithUs("Chat with us"),
+                Section.Support.chatWithUs("Chat with us (\(unreadCount))"),
```
Comment (biafra23) [inline Multisig/UI/Settings/AppSettingsViewController/AppSettingsViewController.swift]: How do I add a badge to the chat icon instead?

File: Multisig/UI/App/MainTabBarViewController.swift (hunk 1 above)
Comment (biafra23) [reply Multisig/UI/App/MainTabBarViewController.swift]: I found `UIColor.button``
File (2)

File: Multisig/UI/Settings/AppSettingsViewController/AppSettingsViewController.swift [hunk 4]
```diff
@@ -195,8 +197,12 @@ class AppSettingsViewController: UITableViewController {
             return tableView.basicCell(name: name, icon: "ico-app-settings-package.pdf", indexPath: indexPath)
             
         case Section.Support.chatWithUs(let name):
-            return tableView.basicCell(name: name, icon: "ico-app-settings-message-circle.pdf", indexPath: indexPath)
-            
+            if Intercom.unreadConversationCount() > 0 {
+                return tableView.basicCell(name: name, icon: "ico-app-settings-support.pdf", indexPath: indexPath)
+            } else {
+                return tableView.basicCell(name: name, icon: "ico-app-settings-message-circle.pdf", indexPath: indexPath)
```
Comment (biafra23) [inline Multisig/UI/Settings/AppSettingsViewController/AppSettingsViewController.swift]: Can we use a different icon here that has a badge?

**Review by sche — APPROVED** [review]

**Review by DmitryBespalov — APPROVED** [review]
👏

**Pull request merged**